builtin/logical/database/backend.go

```diff
@@ -9,13 +9,36 @@ import (
 
 	log "github.com/mgutz/logxi/v1"
 
+	"github.com/hashicorp/errwrap"
+	uuid "github.com/hashicorp/go-uuid"
 	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
 	"github.com/hashicorp/vault/logical"
 	"github.com/hashicorp/vault/logical/framework"
+	"github.com/hashicorp/vault/plugins/helper/database/dbutil"
 )
 
 const databaseConfigPath = "database/config/"
 
+type dbPluginInstance struct {
+	sync.RWMutex
+	dbplugin.Database
+
+	id     string
+	name   string
+	closed bool
+}
+
+func (dbi *dbPluginInstance) Close() error {
+	dbi.Lock()
+	defer dbi.Unlock()
+
+	if dbi.closed {
```

Comment:
Is there anywhere that this is set to true?

Reply:
whoops, :)

Reply:
fixed in https://github.com/hashicorp/vault/pull/3976/commits/41a97688112c6f9dc943eb00edac35d5c7e430c4